persistence/data_models_json_fields.go

```diff
@@ -91,6 +92,119 @@ func (s *StateExecutionSequenceMapsJson) CompleteNewStateExecution(stateId strin
 	return nil
 }
 
+type StateExecutionWaitingQueuesJson struct {
+	// { queue_1: {(state_1, state_1_seq): waiting_count, (state_2, state_2_seq): waiting_count, ...} }
+	QueueToStatesMap map[string]map[string]int `json:"queueToStatesMap"`
+	// { (state_1, state_1_seq): total_waiting_count, (state_2, state_2_seq): total_waiting_count, ...} }
+	StateToQueueCountMap map[string]int `json:"stateToQueueCountMap"`
+}
+
+func NewStateExecutionWaitingQueues() StateExecutionWaitingQueuesJson {
+	return StateExecutionWaitingQueuesJson{
+		QueueToStatesMap:     map[string]map[string]int{},
+		StateToQueueCountMap: map[string]int{},
+	}
+}
+
+func (s *StateExecutionWaitingQueuesJson) ToBytes() ([]byte, error) {
+	return json.Marshal(s)
+}
+
+func NewStateExecutionWaitingQueuesFromBytes(bytes []byte) (StateExecutionWaitingQueuesJson, error) {
+	var waitingQueuesJson StateExecutionWaitingQueuesJson
+	err := json.Unmarshal(bytes, &waitingQueuesJson)
+	return waitingQueuesJson, err
+}
+
+func (s *StateExecutionWaitingQueuesJson) Add(stateExecutionId StateExecutionId, command xdbapi.LocalQueueCommand, anyOfCompletion bool) {
```

Comment:
Nit: `Add` => `AddNewLocalQueueCommandForStateExecution`

Question: what if there are multiple local queue commands?

Reply:
There is a loop to add each local queue command:
```
for _, localQueueCommand := range localQueueCommands {
	waitingQueues.AddNewLocalQueueCommandForStateExecution(request.StateExecutionId, localQueueCommand,
		request.CommandRequest.GetWaitingType() == xdbapi.ANY_OF_COMPLETION)
}
```